hudi-common/src/main/java/org/apache/hudi/common/table/read/UnmergedFileGroupRecordBuffer.java

```diff
@@ -108,6 +122,9 @@ public void processNextDataRecord(BufferedRecord<T> record, Serializable index)
   @Override
   public void processDeleteBlock(HoodieDeleteBlock deleteBlock) {
     // no-op
+    if (emitDelete) {
+      currentInstantLogBlocks.add(deleteBlock);
```

Comment:
was this a bug before?
did we add tests for this

Reply:
looks like it is only used in tests currently.

Reply:
Looks like it is only used in tests, shall we just use the merged reader in the test to get the keys and revert the changes here? The merged reader also supports `emitDelete` I think.

Reply:
Right, it is just the tests and they only read the keys so the merged reader is working. Will revert changes to this file in the next commit.